corehq/apps/accounting/management/commands/change_role_for_software_plan_version.py

```diff
@@ -0,0 +1,89 @@
+import logging
+
+from django.core.management import BaseCommand
+
+from django_prbac.models import Role
+
+from corehq.apps.accounting.models import SoftwarePlanVersion
+
+logger = logging.getLogger(__name__)
+
+
+class OldRoleDoesNotExist(Exception):
+    pass
+
+
+class NewRoleDoesNotExist(Exception):
+    pass
+
+
+def change_role_for_software_plan_version(old_role, new_role, limit_to_plans=None, dry_run=False):
+    """
+    We typically do not support modifying SoftwarePlanVersions directly, and instead encourage creating a new one.
+    This command should only be used when it seems appropriate. The most typical use case would be when it is
+    desirable to delete the Role(slug=old_role) object entirely, and using this command to ensure no software
+    plan versions reference that old role.
+    :param old_role: slug for role to search for
+    :param new_role: slug for role that new software plan version should reference
+    :param limit_to_plans: optionally limit to specific plan names
+    :param dry_run: if False, will make changes to the DB
+    :return: a list of plan names that were modified
+    """
+    dry_run_tag = '[DRY_RUN]' if dry_run else ''
+
+    if not Role.objects.filter(slug=old_role).exists():
+        raise OldRoleDoesNotExist
+
+    try:
+        new_role_obj = Role.objects.get(slug=new_role)
+    except Role.DoesNotExist:
+        raise NewRoleDoesNotExist
+
+    versions = SoftwarePlanVersion.objects.filter(role__slug=old_role, is_active=True)
+    if limit_to_plans:
```

Comment:
do we want this check here if the risk of limiting to plans and then deleting the `Role` means that the `SoftwarePlans` tied to that `Role` also get deleted?

Reply:
Check what exactly? There definitely is that risk, but this command isn't directly tied to deleting a role so it is up to the user of the method to be aware. I could add some docs to the limit_to_plans parameter? What else did you have in mind?

Reply:
Got rid of it! See 20d244ed6981a955ba26e8691d04d11ef80476e6